sdk/interchain/signature-aggregator.go

```diff
@@ -0,0 +1,136 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved
+// See the file LICENSE for licensing terms.
+package interchain
+
+import (
+	"encoding/hex"
+	"fmt"
+	"time"
+
+	"github.com/ava-labs/avalanche-cli/pkg/models"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/message"
+	"github.com/ava-labs/avalanchego/utils/constants"
+	"github.com/ava-labs/avalanchego/utils/logging"
+	apiConfig "github.com/ava-labs/awm-relayer/config"
+	"github.com/ava-labs/awm-relayer/peers"
+	"github.com/ava-labs/awm-relayer/signature-aggregator/aggregator"
+	"github.com/ava-labs/awm-relayer/signature-aggregator/config"
+	"github.com/ava-labs/awm-relayer/signature-aggregator/metrics"
+	awmTypes "github.com/ava-labs/awm-relayer/types"
+	awmUtils "github.com/ava-labs/awm-relayer/utils"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	DefaultQuorumPercentage   = uint64(67)
+	DefaultSignatureCacheSize = uint64(1024 * 1024)
+)
+
+var etnaTime = time.Unix(0, 0)
+
+type SignatureAggregator struct {
+	subnetID         ids.ID
+	quorumPercentage uint64
+	aggregator       *aggregator.SignatureAggregator
+}
+
+func CreateAppRequestNetwork(network models.Network, logLevel logging.Level) (peers.AppRequestNetwork, error) {
+	peerNetwork, err := peers.NewNetwork(
+		logLevel,
+		prometheus.DefaultRegisterer,
+		nil,
+		&config.Config{
+			PChainAPI: &apiConfig.APIConfig{
+				BaseURL: network.Endpoint,
+			},
+			InfoAPI: &apiConfig.APIConfig{
+				BaseURL: network.Endpoint,
+			},
+		},
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create peer network: %w", err)
+	}
+	return peerNetwork, nil
+}
+
+func NewSignatureAggregator(
+	network peers.AppRequestNetwork,
```

Comment:
is it desirable to create the peers.AppRequestNetwork inside this function? so as to minimize funtcions the users are facing with, it seems the peers network is just a required step before creating the aggregator

Reply:
it's done this way to utilize existing mocks from awm-relayer repo. 
let me see if I can abstract it so it will be not user facing

Reply:
done